src/components/TodoItem.tsx

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { Dispatch, FC, FormEvent, useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todo: Todo;
+  deleteTodo?: (todoId: number) => void;
+  onToggle?: (todo: Todo) => void;
+  isLoading: boolean;
+  onRename?: (todo: Todo) => Promise<void>;
+  setLoadingTodosIds?: Dispatch<React.SetStateAction<number[]>>;
+}
+const TodoItem: FC<Props> = ({
+  todo,
+  deleteTodo = () => {},
+  onToggle = () => {},
+  isLoading,
+  onRename = () => {},
+  setLoadingTodosIds = () => {},
+}) => {
+  const [isRenaming, setIsRenaming] = useState(false);
+  const [newTitle, setNewTitle] = useState('');
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedNewTitle = newTitle.trim();
+
+    if (trimmedNewTitle === todo.title) {
+      setIsRenaming(false);
+
+      return;
+    }
+
+    if (!trimmedNewTitle) {
+      deleteTodo(todo.id);
```

Comment:
Remove defaults and just add null checks to the optional props
```suggestion
      deleteTodo?.(todo.id);
```